index.js

```diff
@@ -0,0 +1,50 @@
+
+// 1. Importaciones
+const express = require('express')
+const app = express()
+
+const hbs = require('hbs')
+const PunkAPIWrapper = require('punkapi-javascript-wrapper')
+const punkAPI = new PunkAPIWrapper()
+
+require('dotenv').config()
+
+
+
+// 2.- Middlewares
+app.use(express.static("public"))
+app.set('views', __dirname + '/views')
+app.set('view engine', 'hbs')
+
+hbs.registerPartials(__dirname + '/views/partials')
+
+// 3.- Rutasd
+app.get('/random-beer', (req, res) => {
+    punkAPI
+    .getRandom()
+    .then(beer => {
+      res.render('random-beer', {beer})
+    })
+    .catch(error => console.log(error));
+})
+
+app.get('/beers', (req, res) => {
+    // Una Promise es un objeto que representa la terminación o el fracaso de una operación asíncrona
+    punkAPI
+      .getBeers()
+      .then(beers => res.render('beers', {beers}))
+      .catch(error => console.log(error));
+})
+
+app.get('/', (req, res) => {
+    res.render('home', {home:'home'})
+})
+
```

Comment:
Para hacer la ruta de detalle de una cerveza cada vez que le des click es necesario hacer la ruta, ibas bien en el ancore tag solo que aun no habiamos visto params esta es una alternativa vi que manejaste .then y catch 👍 
```suggestion
app.get('/beers/:beerId', (req, res) => {
  // console.log('params:', req.params);

  punkAPI
    .getBeer(req.params.beerId) // .getBeer() es el método proporcionado por punkAPI
    .then(responseFromApi => {
      // console.log(responseFromApi);
      res.render('beer-details.hbs', { beers: responseFromApi });
    })
    .catch(err => console.log(err));
});
```

Reply:
Y para esto necesitamos otra vista de detalle
```suggestion
<div>
  {{#each beers}}
    <div class="beer">
      {{>beerpartial this}}
      <div class="beer-inner">
        <p>First brewed: {{this.first_brewed}}</p>
        <p>Attenuation level: {{this.attenuation_level}}</p>
        <p>Recommended food pairing:</p>
        <ul>
          {{#each this.food_pairing}}
            <li> {{this}}</li>
          {{/each}}
        </ul>
      </div>
    </div>
  {{/each}}
  <div class="buttons">
    <a class="btn btn-primary button-margin" href="/beers">Go back to beers list</a>
  </div>
</div>
```